modules/azure/frontdoor.go

```diff
@@ -0,0 +1,127 @@
+	fd, err := GetFrontDoorInstanceE(instanceName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return fd
+}
+
+// FrontendEndpointExistsForFrontDoorInstance indicates whether the frontend endpoint instance exists for the Front Door instance.
+// This function would fail the test if there is an error.
+func FrontendEndpointExistsForFrontDoorInstance(t testing.TestingT, endpointName string, instanceName string, resourceGroupName string, subscriptionID string) bool {
+	exists, err := FrontendEndpointExistsE(endpointName, instanceName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return exists
+}
+
+// GetFrontendEndpointsForFrontDoorInstance gets a frontend endpoint by name for a Front Door instance if it exists for the subscription.
+// This function would fail the test if there is an error.
+func GetFrontendEndpointForFrontDoorInstance(t testing.TestingT, endpointName string, instanceName string, resourceGroupName string, subscriptionID string) *frontdoor.FrontendEndpoint {
+	ep, err := GetFrontendEndpointForFrontDoorInstanceE(endpointName, instanceName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return ep
+}
+
+// FrontDoorInstanceExistsE indicates whether the Front Door instance exists and may return an error.
+func FrontDoorInstanceExistsE(instanceName string, resourceGroupName string, subscriptionID string) (bool, error) {
+	_, err := GetFrontDoorInstanceE(instanceName, resourceGroupName, subscriptionID)
+	if err != nil {
+		if ResourceNotFoundErrorExists(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// FrontendEndpointExistsE indicates whether the endpoint exists for the Front Door instance and may return an error.
+func FrontendEndpointExistsE(endpointName string, instanceName string, resourceGroupName string, subscriptionID string) (bool, error) {
+	_, err := GetFrontendEndpointForFrontDoorInstanceE(endpointName, instanceName, resourceGroupName, subscriptionID)
+	if err != nil {
+		if ResourceNotFoundErrorExists(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// GetFrontDoorInstanceE gets the Front Door instance if it exists and may return an error.
+func GetFrontDoorInstanceE(instanceName, resoureGroupName, subscriptionID string) (*frontdoor.FrontDoor, error) {
+	client, err := GetFrontDoorClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	fd, err := client.Get(context.Background(), resoureGroupName, instanceName)
+	if err != nil {
+		return nil, err
+	}
+
+	return &fd, nil
+}
+
+// GetFrontendEndpointForFrontDoorInstanceE gets the Frontend Endpoint for the Front Door instance if it exists and may return an error.
+func GetFrontendEndpointForFrontDoorInstanceE(endpointName, instanceName, resourceGroupName, subscriptionID string) (*frontdoor.FrontendEndpoint, error) {
```

Comment:
consider simplifing the function name.  maybe use
```suggestion
func GetFrontDoorFrontendEndpoinE(endpointName, frontDoorName, resourceGroupName, subscriptionID string) (*frontdoor.FrontendEndpoint, error) {
```